pycparser/c_generator.py

```diff
@@ -72,11 +73,31 @@ def visit_UnaryOp(self, n):
             else:
                 return '%s%s' % (n.op, operand)
 
+    precedence_map = {
+        # Some what of a duplicate of c_generator.CGenerator.precedence
+        '||': 0,
+        '&&': 1,
+        '|': 2,
+        '^': 3,
+        '&': 4,
+        '==': 5, '!=': 5,
+        '>': 6, '>=': 6, '<': 6, '<=': 6,
+        '>>': 7, '<<': 7,
+        '+': 8, '-': 8,
+        '*': 9, '/': 9, '%': 9
+    }
+
     def visit_BinaryOp(self, n):
-        lval_str = self._parenthesize_if(n.left,
-                            lambda d: not self._is_simple_node(d))
-        rval_str = self._parenthesize_if(n.right,
-                            lambda d: not self._is_simple_node(d))
+        lval_str = self._parenthesize_if(
+            n.left,
+            lambda d: not (self._is_simple_node(d) or
```

Comment:
Add a comment explaining the logic here; why you use `>=` for lhs and `>` for rhs, etc.